src/Common/src/TypeSystem/IL/ILImporter.cs

```diff
@@ -24,6 +24,8 @@ internal enum StackValueKind
         NativeInt,
         /// <summary>Any float value.</summary>
         Float,
+        /// <summary>Any double value.</summary>
+        Double,
```

Comment:
Actually, I forgot that this file is shared by other ILImporters. This value is not used by the other ILImporters.

I think it may be better to go with the other option and promote all floats on ILStack to doubles. Mono IL interpretter is on this plan as well.